sqlx-runner/tx.go

```diff
@@ -36,7 +38,7 @@ func WrapSqlxTx(tx *sqlx.Tx) *Tx {
 	if dat.Strict {
 		time.AfterFunc(1*time.Minute, func() {
 			if !newtx.IsRollbacked && newtx.state == txPending {
-				panic("A database transaction was not closed!")
+				fmt.Println("A database transaction was not closed!")
```

Comment:
what is the knock on affect here? as I understand it this mean we'll essentially ignore any timeouts on transactions?

Reply:
If this is the case I think we should improve the messaging as it won't be obvious to the consumer of the log what's happened. Also do we want to set up a monitor so we're aware when this is occurring?

Reply:
Well, it is hard to predict... This panic's effect is that it might take all pods down in one swoop.

We shouldn't probably use a timeout function + panic to "cancel" a query anyways, but then we are not sure if this query will hang or not. Bear in mind that, once it starts panicking it is usually a matter of time for it to kill all pods.

The correct correct way would probably be use a `context.Context` here, but this would require a very big refactor here 🥲

Reply:
About the monitor, it is already created: https://app.datadoghq.com/monitors/108234207

Not sure about what message to have there though :\

Reply:
```suggestion
				fmt.Println("a database transaction is still open after one minute")
```
perhaps? not a huge improvement admittedly as you'd be able to tell that once you'd located the origin of the message

Reply:
what was the effect of the panic on the transaction previously? presumably it would roll it back if the panic was caught by a deferred call to `AutoRollback` or whatever the method was called? perhaps we want to replicate that here and roll back as well as log?

Reply:
Not sure if it ever called the defers, as the panic here happened within a goroutine, which means any outer defers are not called (the program itself panics because it is an unhandled panic).

[You can see a similar behaviour here](https://go.dev/play/p/RESGaj5-MC0). It never calls deferred calls. So probably the transaction for that row keeps dangling or similar.

This is how it looked before it finally had multiple pods dead at the same time:
![Screenshot 2023-01-17 at 12 56 46](https://user-images.githubusercontent.com/97441511/213202648-4152ecaf-39cb-4f2a-b78f-fa53ad7e88fc.png)

Reply:
ah yeah makes sense. Ok cool let's give this a go and see what happens then 👍